google/resource_container_cluster.go

```diff
@@ -589,6 +609,10 @@ func resourceContainerClusterRead(d *schema.ResourceData, meta interface{}) erro
 	}
 	d.Set("node_pool", nps)
 
+	if cluster.IpAllocationPolicy != nil && cluster.IpAllocationPolicy.UseIpAliases {
```

Comment:
Actually I think we should set it no matter what (and this might be a bug elsewhere in the file). If I create a cluster with IP Aliasing, then remove IP Aliasing outside of Terraform, I would want my state to get updated. It's sort of a moot point since IPAllocationPolicy doesn't support update right now, but this will help future-proof it.

Here's what I think `flattenIPAllocationPolicy` should look like:
```
func flattenIPAllocationPolicy(c *container.IPAllocationPolicy) []map[string]interface{} {
	result := []map[string]interface{}{}
	if c != nil && c.UseIpAliases {
		result = append(result, map[string]interface{}{
			"cluster_secondary_range_name":  c.ClusterSecondaryRangeName,
			"services_secondary_range_name": c.ServicesSecondaryRangeName,
		})
	}
	return result
}
```

and then the call to set it would be:
```
	if err := d.Set("ip_allocation_policy", flattenIPAllocationPolicy(cluster.IpAllocationPolicy)); err != nil {
		return err
	}
```

Reply:
You still need to do the nil check, otherwise the code will panic when trying to access the struct's fields (you can prove it by running `TestAccContainerCluster_basic`)

Reply:
I just realized this comment slipped through the cracks and fixed it.

```  gke_nordstrom-k8s-nonprod_us-west1-a_test-lark/default  make testacc TEST=./google TESTARGS='-run=TestAccContainerCluster_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google -v -run=TestAccContainerCluster_basic -timeout 120m
=== RUN   TestAccContainerCluster_basic
--- PASS: TestAccContainerCluster_basic (356.83s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	356.853s
```